addons/website/static/src/js/content/menu.js

```diff
@@ -360,14 +394,28 @@ publicWidget.registry.FixedHeader = BaseAnimatedHeader.extend({
         }
 
         if (this.hiddenOnScrollEl) {
+            let elHeight = 0;
+            if (this.fixedHeader && this.searchbarEl?.matches(".show")) {
+                // Close the dropdown of the search bar if it's open when
+                // scrolling. Otherwise, the calculated height of the
+                // 'hiddenOnScrollEl' element will be incorrect because it will
+                // include the dropdown height.
+                this.searchbarEl.querySelector("input").blur();
```

Comment:
I played a little bit with the search bar and there is a race condition somewhere.
I did your steps and I added some block to be able to scroll
- Scroll a litle bit
- Click on the search input
- Scroll more (the search bar is invisible but I still have the focus on it)
- type "s"
- the page scroll up
Most of the time it's ok but sometimes I have this:
![image](https://github.com/odoo/odoo/assets/78849981/e4a005d8-c4c7-4d27-bea9-9ceca53023bd)
or sometimes this
![image](https://github.com/odoo/odoo/assets/78849981/58110ed6-84b0-4eff-9c0b-b4dc9df14b5c)

Maybe the you should blur the input when it's not visible anymore ?

Reply:
@Guillaume-gdi  Indeed, well spotted, there was an issue...

We weren't waiting for the page to be scrolled to the top before opening the dropdown with the search results. It was actually working "by chance" when the scroll distance wasn't great because we reached the top before the fetch of the results finished. But with your example, where the scroll distance was greater, the dropdown was opening before reaching the top, causing the calculated height of the header to be incorrect at that moment. I searched for several solutions to prevent the dropdown from opening before reaching the top, but I didn't find anything simple. Perhaps I should have added a CSS rule to set the `dropdown-menu` to `display: none` when a certain class was on the `s_searchbar_input` or something like that. Ultimately, I felt that would have been overkill for a use case that shouldn't occur so often. So, I decided to remove the smooth scroll animation and instead scroll instantly. This solves the issue simply.